wgpu-hal/src/metal/adapter.rs

```diff
@@ -111,7 +111,9 @@ impl crate::Adapter for super::Adapter {
 
         // Metal defined pixel format capabilities
         let all_caps = Tfc::SAMPLED_LINEAR
-            | Tfc::STORAGE_WRITE
+            | Tfc::STORAGE_READ_ONLY
+            | Tfc::STORAGE_WRITE_ONLY
+            | Tfc::STORAGE_READ_WRITE
```

Comment:
`all_caps` should not include read-write usage; it should be based on the read-write tier.